omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/VisitDetailsFragmentController.java

```diff
@@ -49,11 +51,23 @@
 
 public class VisitDetailsFragmentController {
 
+   protected List<SimpleObject> cachedEncounters = new ArrayList<SimpleObject>();
+
    public SimpleObject getVisitDetails(@SpringBean("emrApiProperties") EmrApiProperties emrApiProperties,
+         @InjectBeans CoreAppsProperties coreAppsProperties,
          @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService,
          @SpringBean("encounterService") EncounterService encounterService,
-         @RequestParam("visitId") Visit visit, UiUtils uiUtils,
-         UiSessionContext sessionContext) throws ParseException {
+         @RequestParam("visitId") Visit visit,
+         @RequestParam(value="fromEncounter", required=false) Integer encounterIndex,
+         @RequestParam(value="encounterCount", required=false) Integer encounterCount,
+         UiUtils uiUtils,
+         UiSessionContext sessionContext) throws ParseException
+   {
+      if (encounterIndex == null)
```

Comment:
According to the coding conventions at https://wiki.openmrs.org/display/docs/Coding+Conventions we are in favour of using curly braces for single statements. :)